pkg/api/api.go

```diff
@@ -59,6 +66,15 @@ func (cfg *Config) RegisterFlagsWithPrefix(prefix string, f *flag.FlagSet) {
 	f.StringVar(&cfg.PrometheusHTTPPrefix, prefix+"http.prometheus-http-prefix", "/prometheus", "HTTP URL path under which the Prometheus api will be served.")
 }
 
+// Push either wraps the distributor push function as configured or returns the distributor push directly.
+func (cfg *Config) Push(d *distributor.Distributor) push.Func {
```

Comment:
A couple of things:
1. Since it doesn't push, but it wraps the push function, I would call it `wrapDistributorPush()`
2. No need to export it, so name should start with a lowercase letter (eg. `wrapDistributorPush()`)